src/lib/y2partitioner/dialogs/bcache.rb

```diff
@@ -66,75 +68,148 @@ def contents
         end
 
-        # @macro seeAbstractWidget
-        def help
-          "<p>" +
-            # TRANSLATORS: %s stands for name of option
-            format(_(
-                     "%s is the device that will be used as backing device for bcache." \
-                     "It will define the available space of bcache. " \
-                     "The Device will be formatted so any previous content will be wiped out."
-            ), "<b>" + label + "</b>") +
-            "</p>"
+      private
+
+        # sid of the device that is initally selected
+        #
+        # @return [String, nil]
+        def default_sid
+          return nil unless default_device
+
+          default_device.sid.to_s
+        end
+
+        # The first avilable device is selected by default
+        #
+        # @return [Y2Storage::BlkDevice, Y2Storage::BcacheCset, nil] nil if there is no
+        #   devices to select.
+        def default_device
+          devices.first
+        end
+
+        # Selected device
+        #
+        # @return [Y2Storage::BlkDevice, nil] nil if no device has been selected
+        def selected_device
+          return nil if value.nil? || value.empty?
+
+          devices.find { |d| d.sid == value.to_i }
+        end
+
+        def item_for_device(device)
+          label = device.is?(:bcache_cset) ? device.display_name : device.name
+
+          [device.sid.to_s, label]
+        end
+      end
+
+      # Widget to select the backing device
+      class BackingDeviceSelector < DeviceSelector
+        # @return [Y2Storage::BlkDevice, Y2Storage::BcacheCset]
+        attr_reader :caching
+
+        # Constructor
+        #
+        # @param bcache [Y2Storage::Bcache, nil] existing bcache or nil if it is a new one.
+        # @param devices [Array<Y2Storage::BlkDevice>] possible devices that can be used as
+        #   backing device.
+        # @param caching [Y2Partitioner::Dialogs::Bcache::CachingDeviceSelector] reference to
+        #   caching widget. Used to check that caching and backing devices are not identical.
+        def initialize(bcache, devices, caching)
+          super(bcache, devices)
+
+          @caching = caching
         end
 
         # @macro seeAbstractWidget
-        def init
-          return unless @device
+        def label
+          _("Backing Device")
+        end
 
-          self.value = @device.sid.to_s
+        # @macro seeItemsSelection
+        def items
+          devices.map { |d| item_for_device(d) }
         end
 
         # @macro seeAbstractWidget
-        def store
-          val = value
-          @result = @devices.find { |d| d.sid == val.to_i }
+        def help
+          # TRANSLATORS: %{label} is replaced by the label of the option to select a backing device.
+          format(
+            _("<p>" \
+                "<b>%{label}</b> is the device that will be used as backing device for bcache." \
+                "It will define the available space of bcache. " \
```

Comment:
This sentence sounds a little bit strange. What about "It will define the size of the resulting bcache device"?

Reply:
Yes, I prefer your proposal.